documentation/smg/02_development_environment.md

```diff
@@ -1 +1,208 @@
+This description refers to a personal laptop working from home.
+
+### Access
+
+* Using the STFC websites requires use of a VPN.
+There are a number of possible setups; the one described here is apparently not the most proper but works.
+You may already have done this to get email etc.
+* Steps:
+  * From the Windows Start button navigate to Control Panel.
+  * In Control Panel under Category view, select Network and Internet, then Network and Sharing Center.
+  * Click the link Set up a new connection or network.
+  * In the new window, select Connect to a workplace, click Next
+  * Click on the selection Use my Internet connection (VPN)
+  * For Internet Address, use VPN.STFC.AC.UK. For destination name, use CICT VPN.
+  * If you see Next, click on it and fill in your federal ID and its password, with domain CLRC, click Finish.
+  * If you see Finish immediately, go to the Control Panel VPN page.  You will see CICT VPN near the top of the page.
+  Click it to reveal Connect, Advanced options, and Remove button.
+  Click Advanced Options, where you can fill in the ID, password and domain.
+  * Click Connect to establish the connection. This may need refreshing daily, or after shutdown or sleep, or just sometimes.
+  Check back to the VPN page; if the Connect button is visible, you are not connected even if your email is working.
+
+### Getting MATLAB
+
+MATLAB can be found at STFC RAL IT Support (Knowledge Base/MATLAB) at www.facilities.rl.ac.uk/support/KnowledgeBase/Matlab.aspx.
+This should set up an account with Matlab with your STFC email as id and a password of your choice.
+You can then go to the Matlab download page for STFC and download;
+choose R2018b or R2019b, which are the current versions supported at the time of writing.
+Or update to more recent if we have changed. This will give you an installation .exe.
+
+Running the .exe will download the files and install.
+You may be asked for activation details from Matlab.
+This will be your id and password as above.
+You are getting an academic individual STFC licence.
+Activation may also be required on first launch. I
+deally you will be logged in as you with appropriate privileges.
+If not, you may need to run as administrator;
+not doing so puts you in an infinite loop of activate, launch, activate, launch.....
+
+This appears to be a one-user license
+and I am not necessarily expecting that you will be able to install on multiple machines e.g. personal and work machine.
+But it may work.
+
+NB I had problems trying to restart the download - it said I did not have access rights.
+Clearing cookies from the page solved this.
+
+### Using MATLAB
+
+#### Prerequisites
+
+* You need to install `cmake`.
+I installed the current (at time of writing) version 3.18.2;
+there is a minimum version but all current versions should be OK for this.
+Select that `cmake` is on the path.
+* You need to install the Visual Studio SDK.
+This can be the actual SDK or a working VS installation.
```

Comment:
Adding links to the SDK and VS-community download sites would be helpful in this section

Reply:
Done for VS-community. The web is iffy on the C++ SDK - I have mentioned Chocolatey which seems to provide 2017; otherwise the route does seem to be through the full VS installation.

Reply:
The SDK is available from Microsoft: https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2017